dev/src/DocFx/Page/Page.php

```diff
@@ -97,16 +98,69 @@ private function getClassItem(): array
         ]);
     }
 
+    private function toSnakeCase(string $item): string
+    {
+        return strtolower(preg_replace(['/([a-z\d])([A-Z])/', '/([^_])([A-Z][a-z])/'], '$1_$2', $item));
+    }
+
+    /**
+     * If a generated sample exists
+     * (e.g., Asset/samples/V1/AssetServiceClient/analyze_iam_policy.php), replace
+     * the existing inline sample with it.
+     *
+     * TODO(dsupplee): utilize the example annotation in the generated client
+     * method docblocks to link out to the generated samples and remove the
+     * inline ones. This will obviate the need for this method.
+     *
+     * @param string $content
+     * @param string $methodName
+     */
+    private function handleSample(string $content, string $methodName): string
+    {
+        $path = sprintf(
+            '%s/samples/%s/%s.php',
+            $this->componentPath,
+            substr($this->filePath, 0, -4),
+            $this->toSnakeCase($methodName)
+        );
+        $contents = @file_get_contents($path);
+        if ($contents) {
+            // Finds the relevant code between the region tags in the generated sample.
+            if (preg_match('/\/\/ \[START\N*\n(.*?)\/\/ \[END/s', $contents, $match) === 1) {
+                // Generated samples include the method description, which is redundant on the doc
+                // site. This removes the description.
+                $sample = preg_replace(
+                    '/(\/\*\*\n)(.*?)(@param|This sample has been automatically)/s',
+                    '$1 * $3',
+                    $match[1]
+                );
+                // Replaces the existing inline sample with the generated one.
```

Comment:
If this is the preferred sample for the method, why wouldn't we make this change in the GAPIC generator instead?